heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/CacheCoreTestHashIndex.java

```diff
@@ -0,0 +1,478 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsFilter;
+
+import static com.twitter.heron.metricscachemgr.metricscache.MetricsCacheQueryUtils.Granularity.RAW;
+
+public class CacheCoreTestHashIndex {
```

Comment:
Out pattern is to have one test class per src class. So in this case we should have CacheCore and CachCoreTest. Could you consolidate tests to match that pattern.